analytic_engine/src/sst/parquet/async_reader.rs

```diff
@@ -234,93 +178,36 @@ impl<'a> Reader<'a> {
 
     #[cfg(test)]
     pub(crate) async fn row_groups(&mut self) -> Vec<parquet::file::metadata::RowGroupMetaData> {
-        let object_meta = self.storage.head(self.path).await.unwrap();
-        let mut reader = self
-            .reader_factory
-            .create_reader(0, object_meta.into(), None, &ExecutionPlanMetricsSet::new())
-            .unwrap();
-
+        let mut reader =
+            CachableParquetFileReader::new(self.storage.clone(), self.path.clone(), None, None);
         let metadata = reader.get_metadata().await.unwrap();
         metadata.row_groups().to_vec()
     }
 }
 
-impl<'a> Drop for Reader<'a> {
-    fn drop(&mut self) {
-        if self.df_plan.is_none() {
-            return;
-        }
-
-        let df_plan = self.df_plan.take().unwrap();
-        info!(
-            "Reader plan metrics:\n{}",
-            DisplayableExecutionPlan::with_metrics(&*df_plan)
-                .indent()
-                .to_string()
-        );
-    }
-}
-
-#[derive(Debug)]
-pub struct CachableParquetFileReaderFactory {
-    storage: ObjectStoreRef,
-    data_cache: Option<DataCacheRef>,
-}
-
-impl CachableParquetFileReaderFactory {
-    pub fn new(storage: ObjectStoreRef, data_cache: Option<DataCacheRef>) -> Self {
-        Self {
-            storage,
-            data_cache,
-        }
-    }
-}
-
-impl ParquetFileReaderFactory for CachableParquetFileReaderFactory {
-    fn create_reader(
-        &self,
-        partition_index: usize,
-        file_meta: FileMeta,
-        metadata_size_hint: Option<usize>,
-        metrics: &ExecutionPlanMetricsSet,
-    ) -> datafusion::error::Result<Box<dyn AsyncFileReader + Send>> {
-        let parquet_file_metrics =
-            ParquetFileMetrics::new(partition_index, file_meta.location().as_ref(), metrics);
-
-        Ok(Box::new(CachableParquetFileReader::new(
-            self.storage.clone(),
-            self.data_cache.clone(),
-            file_meta.object_meta,
-            parquet_file_metrics,
-            metadata_size_hint,
-        )))
-    }
-}
-
 struct CachableParquetFileReader {
     storage: ObjectStoreRef,
+    path: Path,
     data_cache: Option<DataCacheRef>,
-    meta: ObjectMeta,
-    metrics: ParquetFileMetrics,
-    metadata_size_hint: Option<usize>,
+    file_size: Option<usize>,
+    bytes_scanned: usize,
     cache_hit: usize,
     cache_miss: usize,
 }
 
 impl CachableParquetFileReader {
     fn new(
         storage: ObjectStoreRef,
+        path: Path,
         data_cache: Option<DataCacheRef>,
-        meta: ObjectMeta,
-        metrics: ParquetFileMetrics,
-        metadata_size_hint: Option<usize>,
+        file_size: Option<usize>,
     ) -> Self {
         Self {
             storage,
+            path,
             data_cache,
-            meta,
-            metrics,
-            metadata_size_hint,
+            file_size,
+            bytes_scanned: 0,
```

Comment:
How about moving the three statistics variable into a struct to make the struct clean?